sgx/Makefile

```diff
@@ -0,0 +1,687 @@
+
+Common_Enclave_Link_Flags := $(SGX_COMMON_CFLAGS)
+Common_Enclave_Link_Flags += -Wl,--no-undefined
+Common_Enclave_Link_Flags += -nostdlib
+Common_Enclave_Link_Flags += -nodefaultlibs
+Common_Enclave_Link_Flags += -nostartfiles
+Common_Enclave_Link_Flags += -L$(SGX_SSL_TRUSTED_LIB_PATH)
+Common_Enclave_Link_Flags += -L$(SGX_LIBRARY_PATH)
+Common_Enclave_Link_Flags += -Wl,--whole-archive -lsgx_tsgxssl
+Common_Enclave_Link_Flags += -Wl,--no-whole-archive -lsgx_tsgxssl_crypto
+Common_Enclave_Link_Flags += -Wl,--whole-archive -l$(Trts_Library_Name)
+Common_Enclave_Link_Flags += -Wl,--no-whole-archive -Wl,--start-group
+Common_Enclave_Link_Flags +=   -lsgx_tstdc
+Common_Enclave_Link_Flags +=   -lsgx_tcxx
+Common_Enclave_Link_Flags +=   -lsgx_pthread
+Common_Enclave_Link_Flags +=   -l$(Crypto_Library_Name)
+Common_Enclave_Link_Flags +=   -l$(Service_Library_Name)
+Common_Enclave_Link_Flags += -Wl,--end-group
+Common_Enclave_Link_Flags += -Wl,-Bstatic
+Common_Enclave_Link_Flags += -Wl,-Bsymbolic
+Common_Enclave_Link_Flags += -Wl,-pie,-eenclave_entry
+Common_Enclave_Link_Flags += -Wl,--export-dynamic
+Common_Enclave_Link_Flags += -Wl,--defsym,__ImageBase=0
+Common_Enclave_Link_Flags += -lkmip-sgx
+
+Test_Enclave_Link_Flags := $(Common_Enclave_Link_Flags)
+
+Demo_Enclave_Link_Flags := $(Common_Enclave_Link_Flags)
+Demo_Enclave_Link_Flags += -L/usr/local/lib
+
+Test_Enclave_Name := kmyth_sgx_test_enclave
+Test_Enclave_Lib := $(Test_Enclave_Name).so
+Test_Enclave_Signing_Key := $(Test_Enclave_Name)_private.pem
+Test_Signed_Enclave_Name := $(Test_Enclave_Name).signed.so
+Test_Enclave_Config_File := $(Test_Enclave_Name).config.xml
+
+Demo_Enclave_Name := kmyth_sgx_retrieve_key_demo_enclave
+Demo_Enclave_Lib := $(Demo_Enclave_Name).so
+Demo_Enclave_Signing_Key := $(Demo_Enclave_Name)_private.pem
+Demo_Signed_Enclave_Name := $(Demo_Enclave_Name).signed.so
+Demo_Enclave_Config_File := $(Demo_Enclave_Name).config.xml
+
+ifeq ($(SGX_MODE), HW)
+ifneq ($(SGX_DEBUG), 1)
+ifneq ($(SGX_PRERELEASE), 1)
+Build_Mode = HW_RELEASE
+endif
+endif
+endif
+
+######## Demo Server Build Settings ########
+
+Server_Name := demo/bin/ecdh-server
+Client_Name := demo/bin/ecdh-client
+
+# Message for missing Demo Enclave Signing Key - Fatal Build Error
+define err_no_demo_enclave_signing_key
+FAIL - No Demo Enclave Signing Key found
+Generate or install demo/enclave/$(DemoEnclave_Signing_Key)
+e.g., run 'openssl genrsa -out demo/enclave/$(Demo_Enclave_Signing_Key) -3 3072'
+endef
+
+# Message for missing Test Enclave Signing Key - Fatal Build Error
+define err_no_test_enclave_signing_key
+FAIL - No Test Enclave Signing Key found
+Generate or install test/enclave/$(Test_Enclave_Signing_Key)
+e.g., run 'openssl genrsa -out test/enclave/$(Test_Enclave_Signing_Key) -3 3072'
+endef
+
+.PHONY: test-pre test-all test-run demo-pre demo-all demo-test-keys-certs demo
+test-pre:
+	@mkdir -p test/bin 
+	@if [ ! -f test/enclave/$(Test_Enclave_Signing_Key) ]; then \
```

Comment:
Is there a compelling reason not to use the same singing key for both of the test enclaves?

Reply:
I don't think so, so 2ed25a1 changes it to use the same key for both.